sdks/python/apache_beam/internal/apiclient.py

```diff
@@ -391,18 +391,38 @@ def stage_file(self, gcs_or_local_path, file_name, stream,
 
   # TODO(silviuc): Refactor so that retry logic can be applied.
   @retry.no_retries  # Using no_retries marks this as an integration point.
-  def create_job(self, job):
-    """Submits for remote execution a job described by the workflow proto."""
-    # Stage job resources and add an environment proto with their paths.
+  def create_job(self, job, pipeline_options):
+    """Creates a job description.
+    Additionally, it may stage it and/or submit it for remote execution."""
+    self.create_job_description(job)
+
+    # Stage and submit the job when necessary
+    dataflow_job_file = pipeline_options.view_as(DebugOptions).dataflow_job_file
+    template_location = (
+        pipeline_options.view_as(GoogleCloudOptions).template_location)
+
+    job_location = template_location or dataflow_job_file
+    if job_location:
+      gcs_or_local_path = os.path.dirname(job_location)
+      file_name = os.path.basename(job_location)
+      self.stage_file(gcs_or_local_path, file_name, StringIO(str(job)))
+
+    if not template_location:
+      self.send_job_description()
+
+  def create_job_description(self, job):
+    """Creates a job described by the workflow proto."""
     resources = dependency.stage_job_resources(
         job.options, file_copy=self._gcs_file_copy)
     job.proto.environment = Environment(
         packages=resources, options=job.options,
         environment_version=self.environment_version).proto
     # TODO(silviuc): Remove the debug logging eventually.
     logging.info('JOB: %s', job)
-    request = dataflow.DataflowProjectsJobsCreateRequest()
 
+  def send_job_description(self):
```

Comment:
option B would make sense. `create_job` does not always submit things. And the action is referred as `submit` in other places instead of `send`.

So there will be 3 functions:
`create_job` -> calls `create_job_description` and `submit_job_description`